argus/util.py

```diff
@@ -148,18 +148,34 @@ def run_command_until_condition(self, cmd, cond,
         This method uses and behaves like `run_command_with_retry` but
         with an additional condition parameter.
         """
-        while True:
```

Comment:
Why don't you leave the while loop and decrement the count each time the condition is not met? Try to avoid using recursiveness whenever possible.